lxd/storage/drivers/driver_zfs_utils.go

```diff
@@ -227,7 +271,7 @@ func (d *zfs) setBlocksize(vol Volume, size int64) error {
 		return nil
 	}
 
-	err := d.setDatasetProperties(d.dataset(vol, false), fmt.Sprintf("recordsize=%d", size))
+	err := d.setDatasetProperties(d.dataset(vol, false), false, fmt.Sprintf("recordsize=%d", size))
```

Comment:
This line should probably be in the next commit where you update the usages of `setDatasetProperties`.